packages/flutter/lib/src/widgets/framework.dart

```diff
@@ -18,35 +18,45 @@ export 'package:flutter/foundation.dart' show FlutterError;
 
 // KEYS
 
-/// A Key is an identifier for [Widget]s and [Element]s. A new Widget will only
-/// be used to reconfigure an existing Element if its Key is the same as its
-/// original Widget's Key.
+/// A [Key] is an identifier for [Widget]s and [Element]s.
 ///
-/// Keys must be unique amongst the Elements with the same parent.
+/// A new widget will only be used to update an existing element if its key is
+/// the same as the key of the current widget associted with the element.
 ///
-/// Subclasses of Key should either subclass [LocalKey] or [GlobalKey].
+/// Keys must be unique amongst the [Element]s with the same parent.
+///
+/// Subclasses of [Key] should either subclass [LocalKey] or [GlobalKey].
 abstract class Key {
-  /// Construct a ValueKey<String> with the given String.
+  /// Construct a [ValueKey<String>] with the given [String].
+  ///
   /// This is the simplest way to create keys.
   factory Key(String value) => new ValueKey<String>(value);
 
   /// Default constructor, used by subclasses.
-  const Key.constructor(); // so that subclasses can call us, since the Key() factory constructor shadows the implicit constructor
+  ///
+  /// Useful so that subclasses can call us, because the Key() factory
+  /// constructor shadows the implicit constructor.
+  const Key._();
```

Comment:
that won't work, what if the subclass isn't in this file? (see travis, for instance...)